server/src/main/kotlin/org/javacs/kt/KotlinLanguageServer.kt

```diff
@@ -90,6 +91,16 @@ class KotlinLanguageServer : LanguageServer, LanguageClientAware, Closeable {
         val clientCapabilities = params.capabilities
         config.completion.snippets.enabled = clientCapabilities?.textDocument?.completion?.completionItem?.snippetSupport ?: false
 
+        (params.initializationOptions as? JsonObject)?.get("kotlin")
+            ?.asJsonObject
+            ?.get("gradle")
+            ?.asJsonObject
+            ?.get("home")
+            ?.asString
+            ?.let {
+                config.gradle.home.clear().append(it)
+            }
```

Comment:
Do we need to do this separately from the normal configuration flow in `didChangeConfiguration`? I vaguely recall that there was some discussion upstream (in the LSP repo) on encouraging servers to move to a pull-model where the server fetches configurations through `workspace/configuration` at startup instead of relying on the configuration to always be passed at launch. Ideally we should probably do something similar in the medium term.

Some interesting threads:
- https://github.com/microsoft/vscode-languageserver-node/issues/524
- https://github.com/Microsoft/language-server-protocol/issues/567